core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -15,312 +15,327 @@
-  public static final String FUSE_FS_NAME = "alluxio.fuse.fs.name";
-  public static final String OSS_ACCESS_KEY = "fs.oss.accessKeyId";
-  public static final String OSS_SECRET_KEY = "fs.oss.accessKeySecret";
-  public static final String OSS_ENDPOINT_KEY = "fs.oss.endpoint";
-  public static final String S3N_ACCESS_KEY = "fs.s3n.awsAccessKeyId";
-  public static final String S3N_SECRET_KEY = "fs.s3n.awsSecretAccessKey";
-  // Not prefixed with fs, the s3a property names mirror the aws-sdk property names for ease of use
-  public static final String S3A_ACCESS_KEY = "aws.accessKeyId";
-  public static final String S3A_SECRET_KEY = "aws.secretKey";
-  public static final String GCS_ACCESS_KEY = "fs.gcs.accessKeyId";
-  public static final String GCS_SECRET_KEY = "fs.gcs.secretAccessKey";
-  public static final String SWIFT_USER_KEY = "fs.swift.user";
-  public static final String SWIFT_TENANT_KEY = "fs.swift.tenant";
-  public static final String SWIFT_API_KEY = "fs.swift.apikey";
-  public static final String SWIFT_PASSWORD_KEY = "fs.swift.password";
-  public static final String SWIFT_AUTH_URL_KEY = "fs.swift.auth.url";
-  public static final String SWIFT_AUTH_PORT_KEY = "fs.swift.auth.port";
-  public static final String SWIFT_AUTH_METHOD_KEY = "fs.swift.auth.method";
-  public static final String SWIFT_USE_PUBLIC_URI_KEY = "fs.swift.use.public.url";
-  public static final String SWIFT_SIMULATION = "fs.swift.simulation";
-  // Authentication
-  public static final String SECURITY_AUTHENTICATION_TYPE = "alluxio.security.authentication.type";
-  public static final String SECURITY_AUTHENTICATION_CUSTOM_PROVIDER =
-      "alluxio.security.authentication.custom.provider.class";
-  public static final String SECURITY_AUTHENTICATION_SOCKET_TIMEOUT_MS =
-      "alluxio.security.authentication.socket.timeout.ms";
-  public static final String SECURITY_LOGIN_USERNAME = "alluxio.security.login.username";
-  // Authorization
-  public static final String SECURITY_AUTHORIZATION_PERMISSION_ENABLED =
-      "alluxio.security.authorization.permission.enabled";
-  public static final String SECURITY_AUTHORIZATION_PERMISSION_UMASK =
-      "alluxio.security.authorization.permission.umask";
-  public static final String SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP =
-      "alluxio.security.authorization.permission.supergroup";
-  // Group Mapping
-  public static final String SECURITY_GROUP_MAPPING = "alluxio.security.group.mapping.class";
+  public static boolean isValid(String keyStr) {
+    for (PropertyKey key : PropertyKey.values()) {
+      if (key.toString().equals(keyStr)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Parses a string and return its corresponding {@link PropertyKey}, throwing exception if no such
+   * a property can be found.
+   *
+   * @param keyStr string of property key
+   * @return corresponding property
+   */
+  public static PropertyKey fromString(String keyStr) {
```

Comment:
this method would also benefit from deterministic mapping between enum constants and strings